Update repository

Danielius1922 opened the pull request:
- upgrade go to 1.23
- upgrade github reunners to ubuntu:24.04
- use golangsci-lint v1.61

Upgrade dependencies

Direct:

github.com/plgd-dev/device/v2 v2.5.3-0.20240912113932-acfea60431b9 go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo v0.55.0 go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.55.0 go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.55.0 go.opentelemetry.io/otel v1.30.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.30.0 go.opentelemetry.io/otel/metric v1.30.0
go.opentelemetry.io/otel/sdk v1.30.0
go.opentelemetry.io/otel/trace v1.30.0
golang.org/x/exp v0.0.0-20240909161429-701f63a606c0 golang.org/x/net v0.29.0
google.golang.org/grpc v1.66.2

Indirect:
github.com/go-json-experiment/json v0.0.0-20240815175050-ebd3a8989ca1 github.com/jhump/protoreflect v1.17.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.30.0 golang.org/x/crypto v0.27.0

Danielius1922 commented:
Waiting for go1.23 support in CodeQL https://github.com/github/codeql-action/issues/2467

jkralik commented on go.mod:
```diff
@@ -1,8 +1,8 @@
+go 1.23
 
-toolchain go1.22.0
+toolchain go1.23.0
```
could we leave it on 1.22 - just in go.mod

Danielius1922 replied:
OK, lets keep it at 1.22 for now

Danielius1922 committed:
Update repository

  - upgrade github runners to ubuntu:24.04
  - use golangsci-lint v1.61

Upgrade dependencies

Direct:
github.com/plgd-dev/device/v2 v2.5.3-0.20240916150018-cc07b737d112
github.com/plgd-dev/go-coap/v3 v3.3.5
go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo v0.55.0
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.55.0
go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.55.0
go.opentelemetry.io/otel v1.30.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.30.0
go.opentelemetry.io/otel/metric v1.30.0
go.opentelemetry.io/otel/sdk v1.30.0
go.opentelemetry.io/otel/trace v1.30.0
golang.org/x/net v0.29.0
google.golang.org/grpc v1.66.2

Indirect:
github.com/jhump/protoreflect v1.17.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace
golang.org/x/crypto v0.27.0
github.com/go-json-experiment/json => github.com/go-json-experiment/json v0.0.0-20240815174924-0599f16bf0e2
golang.org/x/exp => golang.org/x/exp v0.0.0-20240823005443-9b4947da3948
